### DISCUSSION TITLE
Use case

### DISCUSSION THREAD

**Discussion by ILLISIS** [Diordany/factorio-ember-autopilot#1] [Ideas]
Hello,

This interests me and I believe I have a very good use case, where there is a good use case there is usually inspiration.

I also struggle at times with the games pathfinder with;

https://mods.factorio.com/mod/Constructron-Continued
https://github.com/ILLISIS/Constructron-Continued

Constructron utilizes both the games pathfinder and a customer A* pathfinder for landfill jobs.

I would be interested in seeing what we could achieve together.

**Comment by Diordany** [comment]
Hi @ILLISIS,

Thank you for reaching out! I'm interested in finding out what use case you're thinking of, and I like the idea of collaboration. I'll also check out the **Constructron** mod later.

Regarding **Ember Autopilot**, I was also leaning towards a custom implementation of the A* algorithm. My next goal is to look into what state space and heuristics might fit my application(s) best. Another long term goal I have for **Ember** is auto-mining and auto-construction, which seems to fall into the same domain as the **Constructron** mod.

**Reply by ILLISIS** [reply]
I would say you're more in the territory of kruise kontrol rather than Constructrons.

The Constructron code isn't in the best shape, it needs a lot of extraction, but the lua A* the pathfinder goes from A > B with what I needed (custom collision checks).

**Reply by Diordany** [reply]
> I would say you're more in the territory of kruise kontrol rather than Constructrons.

Yeah, I can agree with that.

> The Constructron code isn't in the best shape, it needs a lot of extraction, but the lua A* the pathfinder goes from A > B with what I needed (custom collision checks).

So am I correct to assume that you want to collaborate on refining Constructron's pathfinding algorithm? Or is it something else you had in mind?

By the way, I'll join the **Constructron** Discord server later. I think it's easier to communicate there.
